FrontEnd/src/main/java/resourceDisplay/IndexController.java

```diff
@@ -0,0 +1,154 @@
+    public String index() {
+        return "index";
+    }
+
+    /**
+     * This page returns all the assets in the DB
+     *
+     * @param principal - Used to check authentication
+     * @param model     - Used to show table to user
+     * @return the allassets template
+     */
+    @GetMapping("/allassets")
+    public String getAll(@AuthenticationPrincipal OAuth2User principal, Model model) {
+        SpannerOptions options =
+                SpannerOptions.newBuilder().setProjectId(SPANNER_PROJECT_ID).build();
+        spanner = options.getService();
+        db = DatabaseId.of(SPANNER_PROJECT_ID, SPANNER_INSTANCE_ID, SPANNER_DATABASE_ID);
+        dbClient = spanner.getDatabaseClient(db);
+        AssetsRepository assets = new AssetsRepository();
+        //TODO use workspaceID from user/workspace/project
+        ResultListObject resultListObject = assets.getAllAssets(dbClient);
+        model.addAttribute("displayNames", resultListObject.columnDisplays);
+        model.addAttribute("allAssets", resultListObject.columnResults);
+        return "allassets";
+    }
+
+    // todo
+
+    /**
+     * This page returns assets by kind (with the specific data per asset)
+     *
+     * @param principal - Used to check authentication
+     * @param model     - Used to show table to user
+     * @return the bykind template
+     */
+    @GetMapping("/bykind")
+    public String getByKind(@AuthenticationPrincipal OAuth2User principal, Model model,
+                            @ModelAttribute KindObject kindObject) {
+        SpannerOptions options =
+                SpannerOptions.newBuilder().setProjectId(SPANNER_PROJECT_ID).build();
+        spanner = options.getService();
+        db = DatabaseId.of(SPANNER_PROJECT_ID, SPANNER_INSTANCE_ID, SPANNER_DATABASE_ID);
+        dbClient = spanner.getDatabaseClient(db);
+        AssetsRepository assets = new AssetsRepository();
+        model.addAttribute("kindObject", kindObject);
+        //TODO use real workspaceID from user/workspace/project table this is temporary (using "noasan")
+        List<String> kindList = assets.getFilterList(dbClient, "noasan", "kind");
+        model.addAttribute("kindList", kindList);
+        String kind = kindObject.getKind();
+        if (kind != null) {
+            ResultListObject resultListObject = assets.getAssetsByKind(dbClient, kind);
+            model.addAttribute("displayNames", resultListObject.columnDisplays);
+            model.addAttribute("allAssets", resultListObject.columnResults);
+        }
+        return "bykind";
+    }
+
+    /**
+     * This page returns assets by status
+     *
+     * @param principal    - Used to check authentication
+     * @param model        - Used to show table to user
+     * @param statusObject - Used to get filter from user
+     * @return the bystatus template
+     */
+    @GetMapping("/bystatus")
+    public String getByStatus(@AuthenticationPrincipal OAuth2User principal, Model model,
```

Comment:
These methods too share the same logic, so the comment above applies here as well.